src/platform/tests/BUILD.gn

```diff
@@ -39,7 +39,7 @@ if (chip_device_platform != "none" && chip_device_platform != "fake") {
       "${nlunit_test_root}:nlunit-test",
     ]
 
-    if (chip_mdns != "none" && chip_enable_happy_tests &&
+    if (chip_mdns != "none" &&
```

Comment:
Also the reason CI is failing

Reply:
The logic seems off - it looks like enable happy test equates now with 'enable unit test for dnssd' which seems odd to me. I will look into adding a separate flag for this. I am not sure if dnssd were considered "happy tests only" before - I only saw the 5 node multicast code running through happy.

Reply:
Added a separate flag. This is not 1:1 exactly because the test is now enabled in darwin and not in linux. I believe if we enable in linux, we should make sure avahi is running.